replay: alternative structure (#5913)
getsentry/sentry-docs: src/platforms/javascript/common/session-replay/configuration.mdx

description: "Learn about the general Session Replay configuration fields."
---

> ## General Integration Configuration

Author: Starting with the sampling options, which are the main ones/everyone will use

| inlineStylesheet | boolean | `true`  | Should inline stylesheets used in the recording |
| recordCanvas     | boolean | `false` | Should record `<canvas>` elements |

> ## Identifying Users

Author: Identifying user is an SDK specific thing. It'll improve the user experience for sure, so makes total sense to add here. I tried to word it a bit as "here are the actual docs, but here's a quick snippet to get you started"

> You can use the Sentry SDK to set the user of a session, in order to associate a user identity to a session replay, refer to the <PlatformLink to="/enriching-events/identify-user">Identifying User Docs</PlatformLink>.

Author: done in new branch, PR incoming.

Here's a short example using e-mail:

```javascript

> import * as Sentry from "@sentry/browser";
> Sentry.setUser({ email: "sentaur@sentry.dev" });

Reviewer: ```suggestion
Sentry.setUser({ email: "sentaur@sentry.dev" });
```

To spare us another place where we should add platform-specific imports (see my [comment](https://github.com/getsentry/sentry-docs/pull/5899#discussion_r1041993025)), I'd vote to just remove the import here. WDYT?

Author: This is a challenge indeed. We went with "always add imports to snippets" a while back because it happens that users come back with "I get an error of Sentry is undefined" or the equivalent in other languages. But the added overhead of having to maintain multiple snippets is indeed an issue.

Your code comment (in the link you shared) can be a good approach. We can also add a couple of examples like:

Reviewer: Sure, def not a blocker, merge away 👍 
And I agree, we should discuss this in one place and not multiple 😅

  // any further configuration here is passed directly to rrweb
});
```

> ## Start and Stop Recording Manually

Author: I'm assuming here we prefer users keep with defaults and let the SDK start/stop. Is that valid? For that reason, moving to the bottom.

These APIs can become a source of support issues, where the SDK is already start/stopping but users decide to call it too and then are confused by the duration of replays. For that reason, if `debug=true` and `start()` is called but there's already a replay running, worth logging out, for discoverability sake.

Do we know of some use cases when users want to start/stop manually? Is it worth adding here to the docs "you for example want to xyz"?

> Replay recording only starts automatically when it is included in the integrations key when calling `Sentry.init`. Otherwise you can initialize the plugin and manually call the `start()` method on the integration instance. To stop recording you can call the `stop()`.

Reviewer 2: ```suggestion
Replay recording only starts automatically when it's included in the integrations key when calling `Sentry.init`. Otherwise, you can initialize the plugin and manually call the `start()` method on the integration instance. To stop recording you can call the `stop()`:
```

Author: In new branch, PR incoming
